cabal-install/Distribution/Client/CmdInstall.hs

```diff
@@ -155,12 +151,22 @@ newInstallOptions _ =
     "Set the environment file that may be modified."
     ninstEnvironmentPath (\pf flags -> flags { ninstEnvironmentPath = pf })
     (reqArg "ENV" (succeedReadE Flag) flagToList)
-  -- TODO choose a name. --force-overwrite, --overwrite-symlink, --overwrite...
-  , option [] ["force-overwrite"]
-    "Overwrite an existing symlink."
-    ninstForceOverwrite (\v flags -> flags { ninstForceOverwrite = v })
-    trueArg
+  , option [] ["overwrite-policy"]
```

Comment:
We can also make it work similarly to `--allow-newer`, i.e. make `--overwrite-symlink` w/o arguments be a shortcut for `--overwrite-symlink=always`.

Reply:
@23Skidoo btw, didn't we want to deprecate the option-less `--allow-newer` variant? :)

Reply:
oh and btw, I'd like to point out that during dogfooding I started using an abbrev, like e.g.

    cabal v2-install alex --overwrite=always

or even

    cabal v2-install alex --over=always

Reply:
@23Skidoo after thinking about this,  the decision whether to support an-argument less `--overwrite-...` hinges on what default we want to pick. I.e. if we default to `always` it'd be confusing to have a seemingly no-op alias which just reaffirms the default setting.

Reply:
> thus I'd suggest to postpone adding a shortcut to this PR

Sure. Let's merge.